scenario_utility/src/parse.cpp

```diff
@@ -39,36 +39,36 @@ geometry_msgs::Quaternion read_as<geometry_msgs::Quaternion>(const YAML::Node& n
   return quaternion;
 }
 
-DEFINE_READ_ESSENTIAL_SPECIALIZATION(geometry_msgs::Quaternion);
+DEFINE_READ_ESSENTIAL_SPECIALIZATION(geometry_msgs::msg::Quaternion);
 
 template <>
-geometry_msgs::Pose read_as<geometry_msgs::Pose>(const YAML::Node& node)
+geometry_msgs::msg::Pose read_as<geometry_msgs::msg::Pose>(const YAML::Node& node)
 {
-  geometry_msgs::Pose pose {};
+  geometry_msgs::msg::Pose pose {};
 
-  pose.position = read_essential<geometry_msgs::Point>(node, "Position");
-  pose.orientation = read_essential<geometry_msgs::Quaternion>(node, "Orientation");
+  pose.position = read_essential<geometry_msgs::msg::Point>(node, "Position");
+  pose.orientation = read_essential<geometry_msgs::msg::Quaternion>(node, "Orientation");
 
   return pose;
 }
 
-DEFINE_READ_ESSENTIAL_SPECIALIZATION(geometry_msgs::Pose);
+DEFINE_READ_ESSENTIAL_SPECIALIZATION(geometry_msgs::msg::Pose);
 
 template <>
-geometry_msgs::PoseStamped read_as<geometry_msgs::PoseStamped>(const YAML::Node& node)
+geometry_msgs::msg::PoseStamped read_as<geometry_msgs::msg::PoseStamped>(const YAML::Node& node)
 {
-  geometry_msgs::PoseStamped pose_stamped {};
+  geometry_msgs::msg::PoseStamped pose_stamped {};
 
   pose_stamped.header.frame_id = read_optional<std::string>(node, "FrameId", "/map");
-  pose_stamped.header.stamp = ros::Time::now();
-  // pose_stamped.pose = read_essential<geometry_msgs::Pose>(node, "Pose");
-  pose_stamped.pose.position = read_essential<geometry_msgs::Point>(node, "Position");
-  pose_stamped.pose.orientation = read_essential<geometry_msgs::Quaternion>(node, "Orientation");
+  // pose_stamped.header.stamp = ros::Time::now();
```

Comment:
@esteve I was wondering if you knew how to get around this. This is a non-node library so I'm not sure how ROS time should be passed into this package?

Reply:
For now, I would suggest passing clock objects to the function.
If you can check the use case and see if clock objects can be passed easily, that would be great.

Reply:
I checked the code and it seems like this method is not used anywhere else in the code. I can find references to other methods in this package in other packages but not `read_as` which is where the use of ROS time is specified.

Reply:
Hmm... In that case, I think we can ignore with the timestamp.